src/api/spec/models/obs_factory/distribution_spec.rb

```diff
@@ -1,17 +1,229 @@
 require 'rails_helper'
+require 'webmock/rspec'
 
 RSpec.describe ObsFactory::Distribution do
+  let(:project) { create(:project, name: 'openSUSE:Factory') }
+  let(:other_project) { create(:project, name: 'other_project') }
+  let!(:distribution) { ObsFactory::Distribution.new(project) }
+
   describe '::new' do
     def strategy_for(name)
       ObsFactory::Distribution.new(create(:project, name: name)).strategy
     end
 
-    it { expect(strategy_for('openSUSE:Factory')).to         be_kind_of ObsFactory::DistributionStrategyFactory }
+    it { expect(distribution.strategy).to                    be_kind_of ObsFactory::DistributionStrategyFactory }
     it { expect(strategy_for('openSUSE:Factory:PowerPC')).to be_kind_of ObsFactory::DistributionStrategyFactoryPPC }
     it { expect(strategy_for('openSUSE:42.3')).to            be_kind_of ObsFactory::DistributionStrategyOpenSUSE }
     it { expect(strategy_for('SUSE:SLE-12-SP1:GA')).to       be_kind_of ObsFactory::DistributionStrategySLE12SP1 }
     it { expect(strategy_for('SUSE:SLE-15:GA')).to           be_kind_of ObsFactory::DistributionStrategySLE15 }
     it { expect(strategy_for('SUSE:SLE-15-SP1:GA')).to       be_kind_of ObsFactory::DistributionStrategySLE15 }
     it { expect(strategy_for('SUSE:SLE-12-SP3:Update:Products:CASP20')).to be_kind_of ObsFactory::DistributionStrategyCasp }
   end
+
+  describe 'self.attributes' do
+    it { expect(ObsFactory::Distribution.attributes.class).to eq(Array) }
+    it { expect(ObsFactory::Distribution.attributes.count).to eq(13) }
+  end
+
+  describe 'self.find' do
+    context 'with a project' do
+      context 'with distribution' do
+        subject { ObsFactory::Distribution.find(project.name) }
+
+        it { expect(subject.class).to eq(ObsFactory::Distribution) }
```

Comment:
You could use the [be_kind_of](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/type-matchers) matcher.

Reply:
I would add a second test here that ensures that the project was set.